server/src/modules/actions/documents.actions.ts

```diff
@@ -105,6 +105,28 @@ export const getDocumentTypes = async (): Promise<string[]> => {
   });
 };
 
+export const getDocumentColumns = async (type: string): Promise<string[]> => {
+  // get all keys from document content
+  const aggregationPipeline = [
+    { $match: { type_document: type } },
+    { $project: { keys: { $objectToArray: "$content" } } },
+    { $unwind: "$keys" },
+    { $group: { _id: null, uniqueKeys: { $addToSet: "$keys.k" } } },
+    { $unwind: "$uniqueKeys" },
+    { $sort: { uniqueKeys: 1 } }, // Sort keys alphabetically
+  ];
+
+  const result = await getDbCollection("documentContents")
+    .aggregate(aggregationPipeline)
+    .toArray();
```

Comment:
Faut faire attention à ça. Je sais pas la perf de ça.

```suggestion
  const aggregationPipeline = [
    { $match: { type_document: type } },
    { $project: { keys: { $objectToArray: "$content" } } },
    { $unwind: "$keys" },
    { $sort: { "keys.k": 1 } }, // à voir si le sort est mieux avant ou apres le group
    { $group: { _id: "$keys.k" } },
  ];

  const result = await getDbCollection("documentContents")
    .aggregate(aggregationPipeline)
    .toArray();
```

Reply:
sinon je peux trier en `js` après avoir récupéré les colonnes

Reply:
Je penses que ça devrais etre bon comme ça le sort sera pas plus rapide en js de toute maniere